server/rootless.go

```diff
@@ -45,3 +83,27 @@ func makeOCIConfigurationRootless(g *generate.Generator) {
 
 	g.SetLinuxCgroupsPath("")
 }
+
+// getAvailableV2Controllers returns the entries in /sys/fs/cgroup/<SELF>/cgroup.controllers
+func getAvailableV2Controllers() map[string]struct{} {
+	procSelfCgroup, err := cgroups.ParseCgroupFile("/proc/self/cgroup")
+	if err != nil {
+		logrus.WithError(err).Error("failed to parse /proc/self/cgroup")
+		return nil
+	}
+	v2Group := procSelfCgroup[""]
+	if v2Group == "" {
+		return nil
+	}
+	controllersPath := filepath.Join("/sys/fs/cgroup", v2Group, "cgroup.controllers")
+	controllersBytes, err := ioutil.ReadFile(controllersPath)
+	if err != nil {
+		logrus.WithError(err).Errorf("failed to read %s", controllersPath)
+		return nil
+	}
+	result := make(map[string]struct{})
+	for _, controller := range strings.Fields(string(controllersBytes)) {
```

Comment:
I'd use `strings.Split` here as
 - it is faster;
 - the data returned from the kernel do not contain any extra spaces.

Reply:
updated